internal/sign/rsa.go

```diff
@@ -0,0 +1,115 @@
+package sign
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha1" // nolint:gosec
+	"crypto/x509"
+	"encoding/pem"
+	"io"
+	"io/ioutil"
+
+	"github.com/pkg/errors"
+)
+
+// RSASignSHA1Digest signs the provided SHA1 message digest. The key file
+// must be in the PEM format and can either be encrypted or not.
+func RSASignSHA1Digest(sha1Digest []byte, keyFile, passphrase string) ([]byte, error) {
+	if len(sha1Digest) != sha1.Size {
+		return nil, errors.New("digest is not a SHA256 hash")
+	}
+
+	keyFileContent, err := ioutil.ReadFile(keyFile)
+	if err != nil {
+		return nil, errors.Wrap(err, "reading key file")
+	}
+
+	block, _ := pem.Decode(keyFileContent)
+	if block == nil {
+		return nil, errors.New("parse PEM block with private key")
+	}
+
+	blockData := block.Bytes
+	if x509.IsEncryptedPEMBlock(block) {
+		if passphrase == "" {
+			return nil, errors.New("key is encrypted no passphrase provided")
```

Comment:
I would make the error exported, and also I think the wording is a bit off.. maybe `key is encrypted but no passphrase was provided`?

Reply:
Yes, of course. The missing `but` was a typo.